client/render/Canvas.h

```diff
@@ -93,8 +93,8 @@ class Canvas
 	/// renders multiple lines of text with specified parameters
 	void drawText(const Point & position, const EFonts & font, const ColorRGBA & colorDest, ETextAlignment alignment, const std::vector<std::string> & text );
 
-	/// fills selected area with solid color
-	void drawColor(const Rect & target, const ColorRGBA & color);
+	/// fills selected area with solid color, allows replacing or overdrawing

```

Comment:
What is the difference between replace and no replace?

Reply:
replace:
- drawing RGBA 128, 128, 128, 128
- on RGBA 0, 0, 0, 0
- result RGBA  128, 128, 128, 128

overdrawing:
- drawing RGBA 128, 128, 128, 128
- on RGBA 0, 0, 0, 0
- result RGBA  64, 64, 64, 0

Reply:
probably, it can be solved with SDL blending. If this can solve problem, maybe it would be good to take blending mode as an argument instead of unclear bool. Does `Canvas::applyTransparency` do the same?

https://wiki.libsdl.org/SDL2/SDL_BlendMode

Reply:
Maybe possible. But I will have no time the next few days to look into it.

Reply:
Correct. There is method called `SDL_SetSurfaceBlendMode` to do that.

However here separate method makes more sense - for simple "fill surface with color" just SDL_FillRect is enough.

Blended draw is noticeably slower - rather costly creation of new surface, filling it with semi-transparency, and then - blitting one surface onto another (and I think we use screen-sized surfaces here which are quite large)